agent/consul/server_test.go

```diff
@@ -666,14 +666,12 @@ func testVerifyRPC(s1, s2 *Server, t *testing.T) (bool, error) {
 	retry.Run(t, func(r *retry.R) { r.Check(wantPeers(s2, 2)) })
 
 	// Have s2 make an RPC call to s1
-	s2.localLock.RLock()
 	var leader *metadata.Server
-	for _, server := range s2.localConsuls {
+	for _, server := range s2.serverLookup.addressToServer {
```

Comment:
This and the one above are actually racy since you access the internal member w/o the lock. I'd make a new method called `Servers()` that returns a slice of the server metadatas in a safe way and you can use it in both of these places: `len(s2.serverLookup.Servers()) != 2` for the first one, and then `range` over it here.